File: Java/src/com/daily/algothrim/linked/LinkedNode.java
package com.daily.algothrim.linked;

public class LinkedNode<T> {
    public T value;
    public LinkedNode<T> next;

    public LinkedNode(T value, LinkedNode<T> next) {
        this.value = value;
        this.next = next;
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }

    public void setNext(LinkedNode<T> next) {
        this.next = next;
    }

    public LinkedNode<T> getNext() {
        return next;
    }

    public void printAll() {
        System.out.println(value);
        LinkedNode<T> temp = next;
        while (temp != null) {
            System.out.println(temp.value);
            temp = temp.next;
        }
    }
}

File: kotlin/src/com/daily/algothrim/linked/LinkedNode.kt
package com.daily.algothrim.linked

class LinkedNode<T>(var value: T, var next: LinkedNode<T>? = null) {

    fun printAll() {
        println("$value")
        var temp = next
        while (temp != null) {
            println(temp.value)
            temp = temp.next
        }
    }

}